.github/workflows/check-build-test.yml

```diff
@@ -0,0 +1,220 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches: [ master ]
+    tags-ignore: [ v.* ]
+  schedule:
+    - cron:  '0 0 1 * *' # monthly
+
+env:
+  AKKA_TEST_TIMEFACTOR: 10.0
+  EVENT_NAME: ${{ github.event_name }}
+
+jobs:
+  check:
+    name: Check
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          # we don't know what commit the last tag was it's safer to get entire repo so previousStableVersion resolves
+          fetch-depth: 0
```

Comment:
Would be interesting to see if there is middle ground. In other repos we try to fetch the tags explicitly. Not sure what is more efficient, but some repos are quite huge and would benefit from not fetching all history.

Reply:
I didn't experiment with this much.  I tried a depth of 1000 and the dyn ver stuff still failed.

Reply:
We have all kind of workarounds in place for this, e.g.:
https://github.com/lightbend/cloudflow/blob/master/.github/workflows/push.yml#L21

Reply:
I'll see if we need to optimize this later.  Seems fine for now.